examples/qpdfjob-remove-js-action.cc

```diff
@@ -0,0 +1,63 @@
+// This is an example how to use QPDFJob for removing all Javascript and Action in a PDF file
+
+#include <qpdf/QPDFJob.hh>
+#include <qpdf/QPDFUsage.hh>
+#include <qpdf/QUtil.hh>
+
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+
+int
+main(int argc, char* argv[])
+{
+    if (argc != 3) {
+        std::cerr << "Usage: " << argv[0] << " <input file> <output file>\n";
+        return 1;
+    }
+
+    const char* inputFilePath = argv[1];
+    const char* outputFilePath = argv[2];
+
+    try {
+        QPDFJob j;
+        j.config()->inputFile(inputFilePath)->outputFile(outputFilePath)->checkConfiguration();
+        auto qpdf_sp = j.createQPDF();
+        auto& pdf = *qpdf_sp;
+
+        auto names = pdf.getRoot().getKey("/Names");
+        if (names.isDictionary()) {
+            auto js = names.getKey("/JavaScript");
+            if (js.isDictionary()) {
+                auto nested_names = js.getKey("/Names");
+                if (nested_names.isArray()) {
+                    js.replaceKey("/Names", "null"_qpdf);
+                }
+                names.replaceKey("/JavaScript", "null"_qpdf);
+            }
+        }
+
+        for (auto obj: pdf.getAllObjects()) {
+            if (obj.isDictionaryOfType("/Action")) {
+                auto s = obj.getKey("/S");
+                if (s.isName() && s.getName() == "/JavaScript") {
+                    obj.replaceKey("/S", "null"_qpdf);
+                }
```

Comment:
This can be written more compactly as
```cpp
std::string name;
for (auto obj: pdf.getAllObjects()) {
   if (obj.isDictionaryOfType("/Action") && obj.getKey("/S").getValueAsName(name) && 
       name == "/JavaScript") {
          // do whatever you have to
   }
}
```
I assume what you want to do is disable the action. Assuming that an empty string is valid javascript, setting the /JS entry to an empty string may work. Setting the /S entry to null is not valid as this a required entry for action dictionaries, and if you leave this as /JavaScript, /JS is also required. 

You could try to delete the action completely, but then you would have to make sure that the action is removed everywhere where the action is used and a null object is not acceptable, which feel a little bit tricky. My first choice would be to try an empty string.

Reply:
I've updated the code as suggested. But replace /JS with empty string eventually will delete /Action too.

[input.json](https://github.com/user-attachments/files/16321199/input.json)
[output.json](https://github.com/user-attachments/files/16321200/output.json)